src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -881,21 +1514,128 @@ void W3DDisplay::Preload_Texture_Assets(Utf8String texture)
     }
 }
 
+void Create_Bmp_File(const char *filename, const char *data, int width, int height)
+{
+#ifdef BUILD_WITH_D3D8
+    BITMAPINFOHEADER *header = (BITMAPINFOHEADER *)LocalAlloc(LMEM_ZEROINIT, sizeof(BITMAPINFOHEADER));
+    header->biSize = sizeof(BITMAPINFOHEADER);
+    header->biWidth = width;
+    header->biHeight = height;
+    header->biPlanes = 1;
+    header->biBitCount = 24;
+    header->biCompression = 0;
+    header->biSizeImage = 24 * header->biHeight * ((header->biWidth + 7) / 8);
+    header->biClrImportant = 0;
+    HANDLE h = CreateFile(filename, GENERIC_READ | GENERIC_WRITE, 0, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
+
+    if (h != INVALID_HANDLE_VALUE) {
+        BITMAPFILEHEADER fheader;
+        fheader.bfType = 0x4D42;
+        fheader.bfSize = header->biSizeImage + header->biSize + 4 * header->biClrUsed + 14;
+        fheader.bfReserved1 = 0;
+        fheader.bfReserved2 = 0;
+        fheader.bfOffBits = header->biSize + 4 * header->biClrUsed + 14;
```

Comment:
and this
`#define DibNumColors(lpbi)      ((lpbi)->biClrUsed == 0 && (lpbi)->biBitCount <= 8 \
                                    ? (int)(1 << (int)(lpbi)->biBitCount)          \
                                    : (int)(lpbi)->biClrUsed)`
                                    
`hdr.bfOffBits       = (DWORD)sizeof(BITMAPFILEHEADER) + pdib->biSize + DibPaletteSize(pdib);`

Reply:
fixed.